### PR TITLE
Migrate to clap v4

### PR DESCRIPTION
**Pull request by barafael**

### CODE REVIEW

**Review by jayvdb — APPROVED** [review]

File: src/cli.rs
Comment (flomonster) [inline src/cli.rs:1]: We could use clap derive macro instead.

**Review by flomonster — APPROVED** [review]
Great PR

**Comment by jayvdb** [comment]
ping @tbrand

File: src/cli.rs
Comment (jayvdb) [reply src/cli.rs:1]: Suggest that is done as a follow up PR after this one.

**Pull request merged**